src/stores/timed.rs

```diff
@@ -99,30 +101,35 @@ impl<K: Hash + Eq, V> TimedCache<K, V> {
         self.store
             .retain(|_, (instant, _)| instant.elapsed().as_secs() < seconds);
     }
-}
 
-impl<K: Hash + Eq, V> Cached<K, V> for TimedCache<K, V> {
-    fn cache_get<Q>(&mut self, key: &Q) -> Option<&V>
+    fn status<Q>(&mut self, key: &Q) -> Status
     where
         K: std::borrow::Borrow<Q>,
         Q: std::hash::Hash + Eq + ?Sized,
     {
-        let status = {
-            let mut val = self.store.get_mut(key);
-            if let Some(&mut (instant, _)) = val.as_mut() {
-                if instant.elapsed().as_secs() < self.seconds {
-                    if self.refresh {
-                        *instant = Instant::now();
-                    }
-                    Status::Found
-                } else {
-                    Status::Expired
+        let mut val = self.store.get_mut(key);
+        if let Some(&mut (instant, _)) = val.as_mut() {
+            if instant.elapsed().as_secs() < self.seconds {
+                if self.refresh {
+                    *instant = Instant::now();
                 }
+                Status::Found
             } else {
-                Status::NotFound
+                Status::Expired
             }
-        };
-        match status {
+        } else {
+            Status::NotFound
+        }
+    }
+}
+
+impl<K: Hash + Eq, V> Cached<K, V> for TimedCache<K, V> {
+    fn cache_get<Q>(&mut self, key: &Q) -> Option<&V>
+    where
+        K: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq + ?Sized,
+    {
+        match self.status(key) {
```

Comment:
Whole you're refactoring this would you mind updating the get_mut methods to also use this self.status method?